clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -239,7 +239,7 @@
     /** <code> transactional.id </code> */
     public static final String TRANSACTIONAL_ID_CONFIG = "transactional.id";
     public static final String TRANSACTIONAL_ID_DOC = "The TransactionalId to use for transactional delivery. This enables reliability semantics which span multiple producer sessions since it allows the client to guarantee that transactions using the same TransactionalId have been completed prior to starting any new transactions. If no TransactionalId is provided, then the producer is limited to idempotent delivery. " +
-            "Note that <code>enable.idempotence</code> must be enabled if a TransactionalId is configured. " +
+            "If a TransactionalId is configured, <code>enable.idempotence</code> is implied. " +
             "The default is <code>null</code>, which means transactions cannot be used. " +
```

Comment:
`The default is <code>null</code>` - I don't think this is clear enough, is it talking about the default for `TransactionalId` or the default for `enable.idempotence`?